### PR TITLE
Update CupertinoSlidingSegmentedControl control/feedback mechanism

### PR DESCRIPTION
**Pull request by LongCatIsLooong**
## Description

Replace `CupertinoSlidingSegmentedControl.controller` with `groupValue` and `onValueChanged` callback.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?
  
Breaking Change but `CupertinoSlidingSegmentedControl` just got in yesterday.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/test/cupertino/sliding_segmented_control_test.dart [hunk 1]
```diff
@@ -76,26 +95,24 @@ void main() {
...
-            controller: null,
-          ),
+        CupertinoSlidingSegmentedControl<int>(
+          children: children,
+          groupValue: groupValue,
+          onValueChanged: null,
         ),
       );
-      fail('Should not be possible to create segmented control without a controller');
+      fail('Should not be possible to create segmented control without a onValueChanged');
```
Comment (HansMuller) [inline packages/flutter/test/cupertino/sliding_segmented_control_test.dart]: without a => without an

**Review by HansMuller — APPROVED** [review]
LGTM

**Review by justinmc — APPROVED** [review]
LGTM, just some questions for my own education.

What was the motivation behind this PR, just an idea to improve the API?  I agree that it looks simpler this way than with a controller.

File: packages/flutter/lib/src/cupertino/sliding_segmented_control.dart [hunk 2]
```diff
@@ -203,16 +160,58 @@ class CupertinoSlidingSegmentedControl<T> extends StatefulWidget {
...
-  /// if the new selected child widget was tapped on.
-  final ValueNotifier<T> controller;
+  /// ```dart
+  /// class SegmentedControlExample extends StatefulWidget {
+  ///   @override
+  ///   State createState() => SegmentedControlExampleState();
+  /// }
+  ///
+  /// class SegmentedControlExampleState extends State<SegmentedControlExample> {
+  ///   final Map<int, Widget> children = const {
```
[the remark this answers is not in the document]
Comment (HansMuller) [reply packages/flutter/lib/src/cupertino/sliding_segmented_control.dart]: Part of the motivation for this change was to be consistent with the existing [CupertinoSegmentedControl](https://api.flutter.dev/flutter/cupertino/CupertinoSegmentedControl-class.html).

[the remark this answers is not in the document]
Comment (LongCatIsLooong) [reply packages/flutter/lib/src/cupertino/sliding_segmented_control.dart]: I don't think I gave too much thought about it:sweat: , inherited from the original `CupertinoSegmentedControl`. I guess the map keys make reordering (in `didUpdateWidget`) a bit easier, much like `Widget.Key`s?

[the remark this answers is not in the document]
Comment (LongCatIsLooong) [reply packages/flutter/lib/src/cupertino/sliding_segmented_control.dart]: As for the motivation I think we want to keep it as close to `CupertinoSegmetnedControl` as possible, so it looks more familiar and it's easier to migrate from `CupertinoSegmentedControl`.

[the remark this answers is not in the document]
Comment (justinmc) [reply packages/flutter/lib/src/cupertino/sliding_segmented_control.dart]: Sounds good, thanks for the explanations.

**Pull request merged**